Error in object tracking script

```
[2019-09-04 12:02:24] production.ERROR: Error while executing Python script with '/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py /tmp/object_tracking_input_28074.json /tmp/object_tracking_output_28074.json':
Traceback (most recent call last):
  File "/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py", line 82, in <module>
    for keyframe in ObjectTracker(params):
  File "/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py", line 22, in __init__
    self.tracker.init(frame, track_window)
cv2.error: OpenCV(3.4.5) /tmp/opencv-3.4.5/modules/core/src/dxt.cpp:3335: error: (-215:Assertion failed) type == CV_32FC1 || type == CV_32FC2 || type == CV_64FC1 || type == CV_64FC2 in function 'dft'
 {"exception":"[object] (Exception(code: 0): Error while executing Python script with '/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py /tmp/object_tracking_input_28074.json /tmp/object_tracking_output_28074.json':
Traceback (most recent call last):
  File \"/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py\", line 82, in <module>
    for keyframe in ObjectTracker(params):
  File \"/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py\", line 22, in __init__
    self.tracker.init(frame, track_window)
cv2.error: OpenCV(3.4.5) /tmp/opencv-3.4.5/modules/core/src/dxt.cpp:3335: error: (-215:Assertion failed) type == CV_32FC1 || type == CV_32FC2 || type == CV_64FC1 || type == CV_64FC2 in function 'dft'
 at /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php:188)
[stacktrace]
#0 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(97): Biigle\\Modules\\Videos\\Jobs\\TrackObject->python('/var/www/vendor...')
#1 [internal function]: Biigle\\Modules\\Videos\\Jobs\\TrackObject->Biigle\\Modules\\Videos\\Jobs\\{closure}(Object(Biigle\\Modules\\Videos\\Video), '/var/www/storag...')
#2 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(62): call_user_func(Object(Closure), Object(Biigle\\Modules\\Videos\\Video), '/var/www/storag...')
#3 [internal function]: Biigle\\FileCache\\FileCache->Biigle\\FileCache\\{closure}(Array, Array)
#4 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(122): call_user_func(Object(Closure), Array, Array)
#5 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(63): Biigle\\FileCache\\FileCache->batch(Array, Object(Closure))
#6 /var/www/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(221): Biigle\\FileCache\\FileCache->get(Object(Biigle\\Modules\\Videos\\Video), Object(Closure))
#7 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(110): Illuminate\\Support\\Facades\\Facade::__callStatic('get', Array)
#8 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(47): Biigle\\Modules\\Videos\\Jobs\\TrackObject->getTrackingKeyframes(Object(Biigle\\Modules\\Videos\\VideoAnnotation))
#9 [internal function]: Biigle\\Modules\\Videos\\Jobs\\TrackObject->handle()
#10 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#11 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#12 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#13 /var/www/vendor/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#14 /var/www/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(94): Illuminate\\Container\\Container->call(Array)
#15 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(114): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#16 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#17 /var/www/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(98): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#18 /var/www/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(49): Illuminate\\Bus\\Dispatcher->dispatchNow(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject), false)
#19 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(76): Illuminate\\Queue\\CallQueuedHandler->call(Object(Illuminate\\Queue\\Jobs\\RedisJob), Array)
#20 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(320): Illuminate\\Queue\\Jobs\\Job->fire()
#21 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(270): Illuminate\\Queue\\Worker->process('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Illuminate\\Queue\\WorkerOptions))
#22 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(114): Illuminate\\Queue\\Worker->runJob(Object(Illuminate\\Queue\\Jobs\\RedisJob), 'redis', Object(Illuminate\\Queue\\WorkerOptions))
#23 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\\Queue\\Worker->daemon('redis', 'high,default', Object(Illuminate\\Queue\\WorkerOptions))
#24 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(85): Illuminate\\Queue\\Console\\WorkCommand->runWorker('redis', 'high,default')
#25 [internal function]: Illuminate\\Queue\\Console\\WorkCommand->handle()
#26 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#27 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#28 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#29 /var/www/vendor/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#30 /var/www/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\\Container\\Container->call(Array)
#31 /var/www/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#32 /var/www/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#33 /var/www/vendor/symfony/console/Application.php(969): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#34 /var/www/vendor/symfony/console/Application.php(255): Symfony\\Component\\Console\\Application->doRunCommand(Object(Illuminate\\Queue\\Console\\WorkCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#35 /var/www/vendor/symfony/console/Application.php(148): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#36 /var/www/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#37 /var/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(121): Illuminate\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#38 /var/www/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#39 {main}
"} 
```

Another error:
```
[2020-03-19 19:55:24] production.ERROR: Error while executing Python script with '/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py /tmp/object_tracking_input_82845.json /tmp/object_tracking_output_82845.json':
[h264 @ 0x563e29358f40] A non-intra slice in an IDR NAL unit.
[h264 @ 0x563e29358f40] decode_slice_header error
[h264 @ 0x563e2974a4c0] A non-intra slice in an IDR NAL unit.
[h264 @ 0x563e2974a4c0] decode_slice_header error
Traceback (most recent call last):
  File "/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py", line 82, in <module>
    for keyframe in ObjectTracker(params):
  File "/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py", line 22, in __init__
    self.tracker.init(frame, track_window)
cv2.error: OpenCV(3.4.5) /tmp/opencv-3.4.5/modules/core/src/matrix.cpp:466: error: (-215:Assertion failed) 0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows in function 'Mat'
 {"exception":"[object] (Exception(code: 0): Error while executing Python script with '/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py /tmp/object_tracking_input_82845.json /tmp/object_tracking_output_82845.json':
[h264 @ 0x563e29358f40] A non-intra slice in an IDR NAL unit.
[h264 @ 0x563e29358f40] decode_slice_header error
[h264 @ 0x563e2974a4c0] A non-intra slice in an IDR NAL unit.
[h264 @ 0x563e2974a4c0] decode_slice_header error
Traceback (most recent call last):
  File \"/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py\", line 82, in <module>
    for keyframe in ObjectTracker(params):
  File \"/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py\", line 22, in __init__
    self.tracker.init(frame, track_window)
cv2.error: OpenCV(3.4.5) /tmp/opencv-3.4.5/modules/core/src/matrix.cpp:466: error: (-215:Assertion failed) 0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows in function 'Mat'
 at /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php:188)
[stacktrace]
#0 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(97): Biigle\\Modules\\Videos\\Jobs\\TrackObject->python('/var/www/vendor...')
#1 [internal function]: Biigle\\Modules\\Videos\\Jobs\\TrackObject->Biigle\\Modules\\Videos\\Jobs\\{closure}(Object(Biigle\\Modules\\Videos\\Video), '/var/www/storag...')
#2 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(79): call_user_func(Object(Closure), Object(Biigle\\Modules\\Videos\\Video), '/var/www/storag...')
#3 [internal function]: Biigle\\FileCache\\FileCache->Biigle\\FileCache\\{closure}(Array, Array)
#4 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(139): call_user_func(Object(Closure), Array, Array)
#5 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(80): Biigle\\FileCache\\FileCache->batch(Array, Object(Closure))
#6 /var/www/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(245): Biigle\\FileCache\\FileCache->get(Object(Biigle\\Modules\\Videos\\Video), Object(Closure))
#7 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(110): Illuminate\\Support\\Facades\\Facade::__callStatic('get', Array)
#8 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(47): Biigle\\Modules\\Videos\\Jobs\\TrackObject->getTrackingKeyframes(Object(Biigle\\Modules\\Videos\\VideoAnnotation))
#9 [internal function]: Biigle\\Modules\\Videos\\Jobs\\TrackObject->handle()
#10 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array(Array, Array)
#11 /var/www/vendor/laravel/framework/src/Illuminate/Container/Util.php(34): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#12 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#13 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#14 /var/www/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#15 /var/www/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(94): Illuminate\\Container\\Container->call(Array)
#16 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#17 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#18 /var/www/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(98): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#19 /var/www/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(83): Illuminate\\Bus\\Dispatcher->dispatchNow(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject), false)
#20 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\\Queue\\CallQueuedHandler->Illuminate\\Queue\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#21 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#22 /var/www/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(85): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#23 /var/www/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(59): Illuminate\\Queue\\CallQueuedHandler->dispatchThroughMiddleware(Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#24 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(88): Illuminate\\Queue\\CallQueuedHandler->call(Object(Illuminate\\Queue\\Jobs\\RedisJob), Array)
#25 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(348): Illuminate\\Queue\\Jobs\\Job->fire()
#26 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(294): Illuminate\\Queue\\Worker->process('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Illuminate\\Queue\\WorkerOptions))
#27 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(129): Illuminate\\Queue\\Worker->runJob(Object(Illuminate\\Queue\\Jobs\\RedisJob), 'redis', Object(Illuminate\\Queue\\WorkerOptions))
#28 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(112): Illuminate\\Queue\\Worker->daemon('redis', 'high,default', Object(Illuminate\\Queue\\WorkerOptions))
#29 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(96): Illuminate\\Queue\\Console\\WorkCommand->runWorker('redis', 'high,default')
#30 [internal function]: Illuminate\\Queue\\Console\\WorkCommand->handle()
#31 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array(Array, Array)
#32 /var/www/vendor/laravel/framework/src/Illuminate/Container/Util.php(34): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#33 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#34 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#35 /var/www/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#36 /var/www/vendor/laravel/framework/src/Illuminate/Console/Command.php(202): Illuminate\\Container\\Container->call(Array)
#37 /var/www/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#38 /var/www/vendor/laravel/framework/src/Illuminate/Console/Command.php(189): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#39 /var/www/vendor/symfony/console/Application.php(1011): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#40 /var/www/vendor/symfony/console/Application.php(272): Symfony\\Component\\Console\\Application->doRunCommand(Object(Illuminate\\Queue\\Console\\WorkCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#41 /var/www/vendor/symfony/console/Application.php(148): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#42 /var/www/vendor/laravel/framework/src/Illuminate/Console/Application.php(90): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#43 /var/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(131): Illuminate\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#44 /var/www/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#45 {main}
"} 
```

A new error (video 1699):

```
[2020-04-14 05:34:58] production.ERROR: Error while executing Python script with '/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py /tmp/object_tracking_input_89448.json /tmp/object_tracking_output_89448.json':
Traceback (most recent call last):
  File "/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py", line 82, in <module>
    for keyframe in ObjectTracker(params):
  File "/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py", line 22, in __init__
    self.tracker.init(frame, track_window)
cv2.error: OpenCV(3.4.5) /tmp/opencv-3.4.5/modules/imgproc/src/resize.cpp:3784: error: (-215:Assertion failed) !ssize.empty() in function 'resize'
 {"exception":"[object] (Exception(code: 0): Error while executing Python script with '/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py /tmp/object_tracking_input_89448.json /tmp/object_tracking_output_89448.json':
Traceback (most recent call last):
  File \"/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py\", line 82, in <module>
    for keyframe in ObjectTracker(params):
  File \"/var/www/vendor/biigle/videos/src/config/../resources/scripts/ObjectTracker.py\", line 22, in __init__
    self.tracker.init(frame, track_window)
cv2.error: OpenCV(3.4.5) /tmp/opencv-3.4.5/modules/imgproc/src/resize.cpp:3784: error: (-215:Assertion failed) !ssize.empty() in function 'resize'
 at /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php:188)
[stacktrace]
#0 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(97): Biigle\\Modules\\Videos\\Jobs\\TrackObject->python('/var/www/vendor...')
#1 [internal function]: Biigle\\Modules\\Videos\\Jobs\\TrackObject->Biigle\\Modules\\Videos\\Jobs\\{closure}(Object(Biigle\\Modules\\Videos\\Video), '/var/www/storag...')
#2 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(79): call_user_func(Object(Closure), Object(Biigle\\Modules\\Videos\\Video), '/var/www/storag...')
#3 [internal function]: Biigle\\FileCache\\FileCache->Biigle\\FileCache\\{closure}(Array, Array)
#4 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(139): call_user_func(Object(Closure), Array, Array)
#5 /var/www/vendor/biigle/laravel-file-cache/src/FileCache.php(80): Biigle\\FileCache\\FileCache->batch(Array, Object(Closure))
#6 /var/www/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Biigle\\FileCache\\FileCache->get(Object(Biigle\\Modules\\Videos\\Video), Object(Closure))
#7 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(110): Illuminate\\Support\\Facades\\Facade::__callStatic('get', Array)
#8 /var/www/vendor/biigle/videos/src/Jobs/TrackObject.php(47): Biigle\\Modules\\Videos\\Jobs\\TrackObject->getTrackingKeyframes(Object(Biigle\\Modules\\Videos\\VideoAnnotation))
#9 [internal function]: Biigle\\Modules\\Videos\\Jobs\\TrackObject->handle()
#10 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array(Array, Array)
#11 /var/www/vendor/laravel/framework/src/Illuminate/Container/Util.php(36): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#12 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#13 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#14 /var/www/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#15 /var/www/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(94): Illuminate\\Container\\Container->call(Array)
#16 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#17 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#18 /var/www/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(98): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#19 /var/www/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(83): Illuminate\\Bus\\Dispatcher->dispatchNow(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject), false)
#20 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\\Queue\\CallQueuedHandler->Illuminate\\Queue\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#21 /var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#22 /var/www/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(85): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#23 /var/www/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(59): Illuminate\\Queue\\CallQueuedHandler->dispatchThroughMiddleware(Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Biigle\\Modules\\Videos\\Jobs\\TrackObject))
#24 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(88): Illuminate\\Queue\\CallQueuedHandler->call(Object(Illuminate\\Queue\\Jobs\\RedisJob), Array)
#25 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(368): Illuminate\\Queue\\Jobs\\Job->fire()
#26 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(314): Illuminate\\Queue\\Worker->process('redis', Object(Illuminate\\Queue\\Jobs\\RedisJob), Object(Illuminate\\Queue\\WorkerOptions))
#27 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(134): Illuminate\\Queue\\Worker->runJob(Object(Illuminate\\Queue\\Jobs\\RedisJob), 'redis', Object(Illuminate\\Queue\\WorkerOptions))
#28 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(112): Illuminate\\Queue\\Worker->daemon('redis', 'high,default', Object(Illuminate\\Queue\\WorkerOptions))
#29 /var/www/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(96): Illuminate\\Queue\\Console\\WorkCommand->runWorker('redis', 'high,default')
#30 [internal function]: Illuminate\\Queue\\Console\\WorkCommand->handle()
#31 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array(Array, Array)
#32 /var/www/vendor/laravel/framework/src/Illuminate/Container/Util.php(36): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#33 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#34 /var/www/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#35 /var/www/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#36 /var/www/vendor/laravel/framework/src/Illuminate/Console/Command.php(134): Illuminate\\Container\\Container->call(Array)
#37 /var/www/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#38 /var/www/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#39 /var/www/vendor/symfony/console/Application.php(1001): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#40 /var/www/vendor/symfony/console/Application.php(271): Symfony\\Component\\Console\\Application->doRunCommand(Object(Illuminate\\Queue\\Console\\WorkCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#41 /var/www/vendor/symfony/console/Application.php(147): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#42 /var/www/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#43 /var/www/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(131): Illuminate\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#44 /var/www/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#45 {main}
"} 
```

--- a/src/Rules/VideoUrl.php
+++ b/src/Rules/VideoUrl.php
@@ -4,7 +4,10 @@ namespace Biigle\Modules\Videos\Rules;
 
 use App;
 use Storage;
+use FileCache;
+use FFMpeg\FFProbe;
 use GuzzleHttp\Client;
+use Biigle\FileCache\GenericFile;
 use Illuminate\Contracts\Validation\Rule;
 use GuzzleHttp\Exception\ServerException;
 use GuzzleHttp\Exception\ClientException;
@@ -27,6 +30,13 @@ class VideoUrl implements Rule
     protected $allowedMimes;
 
     /**
+     * Allowed video codecs
+     *
+     * @var array
+     */
+    protected $allowedCodecs;
+
+    /**
      * Create a new instance.
      */
     public function __construct()
@@ -37,6 +47,12 @@ class VideoUrl implements Rule
             'video/mp4',
             'video/quicktime',
             'video/webm',
+        ];
+        $this->allowedCodecs = [
+            'h264',
+            'vp8',
+            'vp9',
+            'av1',
         ];
     }
 
@@ -97,7 +113,13 @@ class VideoUrl implements Rule
             return false;
         }
 
-        return $this->passesMimeType(Storage::disk($disk)->mimeType($path));
+        if (!$this->passesMimeType(Storage::disk($disk)->mimeType($path))) {
+            return false;
+        }
+
+        return FileCache::getOnce(new GenericFile($value), function ($file, $path) {
+            return $this->passesCodec($path);
+        });
     }
 
     /**
@@ -128,8 +150,11 @@ class VideoUrl implements Rule
         }
 
         $mime = explode(';', $response->getHeaderLine('Content-Type'))[0];
+        if (!$this->passesMimeType($mime)) {
+            return false;
+        }
 
-        return $this->passesMimeType($mime);
+        return $this->passesCodec($value);
     }
 
     /**
@@ -143,6 +168,26 @@ class VideoUrl implements Rule
     {
         if (!in_array($mime, $this->allowedMimes)) {
             $this->message = "Videos of type '{$mime}' are not supported.";
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate the codec of a video
+     *
+     * @param string $url URL/path to the video file
+     *
+     * @return bool
+     */
+    protected function passesCodec($url)
+    {
+        $codec = FFProbe::create()->streams($url)->videos()->first()->get('codec_name');
+
+        if (!in_array($codec, $this->allowedCodecs)) {
+            $this->message = "Videos with codec '{$codec}' are not supported.";
 
             return false;
         }

--- a/src/Rules/VideoUrl.php
+++ b/src/Rules/VideoUrl.php
@@ -5,13 +5,13 @@ namespace Biigle\Modules\Videos\Rules;
 use App;
 use Storage;
 use FileCache;
-use FFMpeg\FFProbe;
 use GuzzleHttp\Client;
 use Biigle\FileCache\GenericFile;
 use Illuminate\Contracts\Validation\Rule;
 use GuzzleHttp\Exception\ServerException;
 use GuzzleHttp\Exception\ClientException;
 use GuzzleHttp\Exception\RequestException;
+use Biigle\Modules\Videos\Support\VideoCodecExtractor;
 
 class VideoUrl implements Rule
 {
@@ -184,7 +184,7 @@ class VideoUrl implements Rule
      */
     protected function passesCodec($url)
     {
-        $codec = FFProbe::create()->streams($url)->videos()->first()->get('codec_name');
+        $codec = App::make(VideoCodecExtractor::class)->extract($url);
 
         if (!in_array($codec, $this->allowedCodecs)) {
             $this->message = "Videos with codec '{$codec}' are not supported.";

--- a/src/Support/VideoCodecExtractor.php
+++ b/src/Support/VideoCodecExtractor.php
@@ -1,0 +1,31 @@
+<?php
+
+namespace Biigle\Modules\Videos\Support;
+
+use FFMpeg\FFProbe;
+
+class VideoCodecExtractor
+{
+    /**
+     * FFProbe instance.
+     *
+     * @var FFProbe
+     */
+    protected $ffprobe;
+
+    /**
+     * Extract the video codec from the specified video file.
+     *
+     * @param string $url
+     *
+     * @return string
+     */
+    public function extract($url)
+    {
+        if (!isset($this->ffprobe)) {
+            $this->ffprobe = FFProbe::create();
+        }
+
+        return $this->ffprobe->streams($url)->videos()->first()->get('codec_name');
+    }
+}

--- a/src/resources/views/show/sidebar-edit.blade.php
+++ b/src/resources/views/show/sidebar-edit.blade.php
@@ -26,7 +26,7 @@
                     @if (config('biigle.offline_mode'))
                       The video file on the BIIGLE server (e.g. <code>local://videos/file.mp4</code>).
                    @else
-                      The video file of a <a href="{{route('manual-tutorials', ['videos', 'remote-videos'])}}">remote video</a> (e.g. <code>https://my-domain.tld/videos/file.mp4</code>) or on the BIIGLE server (e.g. <code>local://videos/file.mp4</code>).
+                      The video file of a <a href="{{route('manual-tutorials', ['videos', 'remote-videos'])}}">remote video</a> (e.g. <code>https://my-domain.tld/videos/file.mp4</code>) or on the BIIGLE server (e.g. <code>local://videos/file.mp4</code>). Supported video file formats are: MP4 (H.264) and WebM (VP8, VP9, AV1).
                    @endif
                 </p>
             @endif

--- a/src/resources/views/store.blade.php
+++ b/src/resources/views/store.blade.php
@@ -34,7 +34,7 @@
                 @if (config('biigle.offline_mode'))
                   The video file on the BIIGLE server (e.g. <code>local://videos/file.mp4</code>).
                @else
-                  The video file of a <a href="{{route('manual-tutorials', ['videos', 'remote-videos'])}}">remote video</a> (e.g. <code>https://my-domain.tld/videos/file.mp4</code>) or on the BIIGLE server (e.g. <code>local://videos/file.mp4</code>). Supported video file formats are: MP4 (H.264) and WebM.
+                  The video file of a <a href="{{route('manual-tutorials', ['videos', 'remote-videos'])}}">remote video</a> (e.g. <code>https://my-domain.tld/videos/file.mp4</code>) or on the BIIGLE server (e.g. <code>local://videos/file.mp4</code>). Supported video file formats are: MP4 (H.264) and WebM (VP8, VP9, AV1).
                @endif
             </p>
             @if($errors->has('url'))

--- a/tests/Http/Controllers/Api/ProjectVideoControllerTest.php
+++ b/tests/Http/Controllers/Api/ProjectVideoControllerTest.php
@@ -14,6 +14,7 @@ use Biigle\Modules\Videos\Video;
 use Biigle\Modules\Videos\Project;
 use GuzzleHttp\Handler\MockHandler;
 use Biigle\Modules\Videos\Jobs\ProcessNewVideo;
+use Biigle\Modules\Videos\Support\VideoCodecExtractor;
 
 class ProjectVideoControllerTest extends ApiTestCase
 {
@@ -106,12 +107,20 @@ class ProjectVideoControllerTest extends ApiTestCase
             return $client;
         });
 
+        $this->app->bind(VideoCodecExtractor::class, function () {
+            return new class {
+                public function extract($url) {
+                    return 'h264';
+                }
+            };
+        });
+
         $id = $this->project()->id;
         $project = Project::find($id);
         $this->beAdmin();
         $this->json('POST', "/api/v1/projects/{$id}/videos", [
                 'name' => 'my video no. 2',
-                'url' => 'https://domain.tdl/video.mp4',
+                'url' => 'https://domain.tld/video.mp4',
             ])
             ->assertSuccessful();
 

--- a/tests/Rules/VideoUrlTest.php
+++ b/tests/Rules/VideoUrlTest.php
@@ -13,6 +13,7 @@ use GuzzleHttp\Psr7\Response;
 use GuzzleHttp\Handler\MockHandler;
 use Biigle\Modules\Videos\Rules\VideoUrl;
 use GuzzleHttp\Exception\RequestException;
+use Biigle\Modules\Videos\Support\VideoCodecExtractor;
 
 class VideoUrlTest extends TestCase
 {
@@ -132,6 +133,14 @@ class VideoUrlTest extends TestCase
         $client = new Client(['handler' => $handler]);
         $this->app->bind(Client::class, function () use ($client) {
             return $client;
+        });
+
+        $this->app->bind(VideoCodecExtractor::class, function () {
+            return new class {
+                public function extract($url) {
+                    return 'h264';
+                }
+            };
         });
 
         $validator = new VideoUrl;

--- a/tests/Support/VideoCodecExtractorTest.php
+++ b/tests/Support/VideoCodecExtractorTest.php
@@ -1,0 +1,15 @@
+<?php
+
+namespace Biigle\Tests\Modules\Videos\Support;
+
+use TestCase;
+use Biigle\Modules\Videos\Support\VideoCodecExtractor;
+
+class VideoCodecExtractorTest extends TestCase
+{
+    public function testExtract()
+    {
+        $extractor = new VideoCodecExtractor;
+        $this->assertEquals('h264', $extractor->extract(__DIR__.'/../files/test.mp4'));
+    }
+}

